docs/grammar/Solidity.g4

```diff
@@ -189,23 +190,43 @@ locals[
 	(Semicolon | body=block);
 
 /**
- * Definitions of the special fallback and receive functions.
+ * Definition of the special fallback function.
  */
-fallbackReceiveFunctionDefinition
+fallbackFunctionDefinition
 locals[
-	boolean visibilitySet = false,
 	boolean mutabilitySet = false,
 	boolean virtualSet = false,
-	boolean overrideSpecifierSet = false
+	boolean overrideSpecifierSet = false,
+	boolean hasParameters = false
 ]
 :
-	kind=(Fallback | Receive) LParen RParen
+	kind=Fallback LParen (parameterList { $hasParameters = true; } )? RParen
 	(
-		{!$visibilitySet}? visibility {$visibilitySet = true;}
+		{!$visibilitySet}? External {$visibilitySet = true;}
```

Comment:
Since external visibility is unconditional, `visibilitySet` is not required. Perhaps this will also make the failing antlr CI pass

Reply:
It is unconditional, but cannot be specified twice.

Reply:
Ah I see!